SampleApp/SampleApp/SampleApp.swift

```diff
@@ -10,31 +10,74 @@ import TyroTapToPaySDK
+  @State var loadingState: LoadingState = .inProgress("Initialising...")
+  @State var transactionOutcome: TransactionOutcome?
+  @State var opacity: Float = .zero
 
   private let tyroEnvironment = TyroEnvironment.sandbox
   private let connectionProvider: ConnectionProvider
   private let tapToPaySDK: TyroTapToPay
+  private let posInformation = POSInformation(name: "SampleApp POS",
+                                              vendor: "Tyro",
+                                              version: "0.3.0",
+                                              siteReference: "Sydney")
 
   init() {
     let restClient = TyroRestClient(environment: tyroEnvironment)
-    self.connectionProvider = SandboxConnectionProvider(restClient: restClient)
+    connectionProvider = SandboxConnectionProvider(restClient: restClient)
     do {
-      self.tapToPaySDK = try TyroTapToPay(environment: tyroEnvironment,
-                                          connectionProvider: connectionProvider)
-    } catch {
+      tapToPaySDK = try TyroTapToPay(environment: .sandbox, connectionProvider: connectionProvider)
+    }
+    catch {
       fatalError(error.localizedDescription)
     }
   }
 
   var body: some Scene {
     WindowGroup {
-      PaymentsView(viewModel: PaymentsViewModel(tapToPaySDK: tapToPaySDK))
-    }
-    .onChange(of: scenePhase) { (_, newValue) in
-      notificationCentre.post(name: Notification.Name(rawValue: "didChangeScenePhase"),
-                              object: scenePhase)
+      switch loadingState {
+      case .inProgress, .failure:
+        LoadingView(loadingState: $loadingState)
+          .task {
+            withAnimation(.easeIn(duration: 1)) {
+              opacity = 1.0
+            }
+            do {
+              try await tapToPaySDK.connect()
+            } catch {
+              loadingState = .failure(error)
+            }
+          }
+      case .ready:
+        PaymentsView() { (transactionType, amount) in
+          transactionOutcome = try await startTransaction(type: transactionType, amount: amount)
+        }
+      }
+    }.onChange(of: scenePhase) { (_, scenePhase) in
+        // TODO: - notify SDK of change in foreground/background state.
+      }
+  }
+
+  private func startTransaction(type transactionType: TransactionType,
+                                amount: String) async throws -> TransactionOutcome {
+    let transactionDetail = transform(amount)
+    switch transactionType {
+    case .payment:
+      return try await tapToPaySDK.startPayment(transactionDetail: transactionDetail)
+    case .refund:
+      return try await tapToPaySDK.refundPayment(transactionDetail: transactionDetail)
     }
   }
+
+  private func transform(_ amount: String) -> TransactionDetail {
+    TransactionDetail(amount: amount,
+                      referenceNumber: UUID().uuidString, // TODO: determine if/how this should be generated.
+                      transactionID: UUID().uuidString,
```

Comment:
FYI from the ss docs: Setting the sales/refund transaction ID value for the Void transaction, returned from the MPOS server during the Sale/Refund transaction.

Reply:
This can be fixed in another pr as its minor tho

Reply:
Yeah, ideally we could populate a list with payment transactions and store the outcome DTOs and then select one for a refund workflow and then we would have access to the transaction reference. I created a seperate task to look into that for another PR.